### DISCUSSION TITLE
Replicating NetBox - Server Error

### DISCUSSION THREAD

**Discussion by michealsh2020** [netbox-community/netbox#13702] [Q&A]
After following the below 
https://docs.netbox.dev/en/stable/administration/replicating-netbox/ 
Can't log in - how to fix this? 

Server Error
A database programming error was detected while processing this request. Common causes include the following:

Database migrations missing. When upgrading to a new NetBox release, the upgrade script must be run to apply any new database migrations. You can run migrations manually by executing python3 manage.py migrate from the command line.

Unsupported PostgreSQL version. Ensure that PostgreSQL version 12 or later is in use. You can check this by connecting to the database using NetBox's credentials and issuing a query for SELECT VERSION().

The complete exception is provided below:

<class 'django.db.utils.ProgrammingError'>

relation "users_userconfig" does not exist
LINE 1: ...config"."user_id", "users_userconfig"."data" FROM "users_use...
                                                             ^

Python version: 3.10.12
NetBox version: 3.6.0
Plugins: None installed

**Comment by awfki** [comment]
You don't say what you've tried.

- Is the new server running "PostgreSQL version 12 or later"?
- Did you try `python3 manage.py migrate` from the command line?

I have two NetBox VMs and the instructions you linked are the ones I follow to move the DB over.   I dump the database from the active box, copy the file to the standby, then drop the standby's database and import the one I just copied.

Be really really careful about which box you're on unless you want to drop the active database.

**Comment by candlerb** [comment]
Looks like you simply haven't restored the database correctly into the new instance (i.e. all the tables are missing)

**Reply by candlerb** [reply]
I just thought, what version of Netbox is running on the original database (the source of the pgdump?)

If it's earlier than 3.6.0, you will need to run `./upgrade.sh` on the target system after restoring the database.

If it's earlier than 2.11.0, then it won't work at all.  You first have to restore in a 2.11.12 system, run `./upgrade.sh`, *then* it's safe to dump into a 3.6.0 system (or do an in-place upgrade to 3.6.0), and run `./upgrade.sh` again.

Otherwise: to check if the restore was successful: login to the target postgres using `psql` and the same credentials as in configuration.py, and look for the tables using `\dt`.  Compare this with the same command run on the target system.
